clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java

```diff
@@ -249,9 +249,9 @@ public String getFileLocation(NameIdentifier ident, String subPath, String locat
       CallerContext callerContext = CallerContext.CallerContextHolder.get();
 
       Map<String, String> params = new HashMap<>();
-      params.put("sub_path", RESTUtils.encodeString(subPath));
```

Comment:
Java and Python http client will help to encode the query parameters automatically, doing this manually will introduce the double encoding problem.